### PR TITLE
feat(CI): use prod build for integration tests

### PR DESCRIPTION
**Pull request by mattkrick**
This PR runs the integration tests against a production build vs. a development build.
This is beneficial for a few reasons:
- it's closer to production, so we may catch an extra bug at some point
- the CI already performs a prod build, so CI goes faster because it no longer has to run a dev build
- the dev build in #6337 includes a bunch of watchers which break in CI. We could make it work, but given the above 2 benefits, this seems like the best solution

Also deleted a handful of unused files.

TEST
- [ ] the tests in CI pass

### CODE REVIEW

File: packages/server/__tests__/common.ts [hunk 1]
```diff
@@ -30,13 +32,27 @@ export async function sendIntranet(req: {
   return response.json()
 }
 
+const persistQuery = async (query: string) => {
+  const r = await getRethink()
+  const docId = persistFunction(query.trim())
```
Comment (Dschoordsch) [inline packages/server/__tests__/common.ts]: -2 `await` is missing

**Comment by blimmer** [comment]
See also https://github.com/ParabolInc/parabol/issues/6381, which we should consider as part of this PR

File: packages/server/database/rethinkDriver.ts [hunk 2]
```diff
@@ -231,7 +231,11 @@ const getRethink = async () => {
   }
   // this is important because pm2 will restart the process & for whatever reason r isn't always healthy
   await r.waitForHealthy()
-  return (r as unknown) as ParabolR
+  return r as unknown as ParabolR
+}
+
+export const closeRethink = async () => {
+  await (await promise)?.drain()
```
Comment (mattkrick) [inline packages/server/database/rethinkDriver.ts:238]: little nit, but we'll want to drain the whole pool, not just the connection (r.getPoolMaster()?.drain())

**Comment by mattkrick** [comment]
i don't know what gypsy magic jest is doing, but the connection pool started in the tests is not the same one that is started in globalTeardown. They have the same pid & ppid, so some kind of sandboxing is going on. 

Jest says the following:
```
Note: Any global variables that are defined through globalSetup can only be read in globalTeardown. You cannot retrieve globals defined here in your test suites.
```

Anyways, the "right" way to do this is to close down the DB after the tests in a single file are complete, so that's what i did.
in the future we'll reuse this `afterAll` pattern to clean up certain tables probably, so it's really just introducing the pattern a little early.

File: packages/server/database/migrations/20180206135107-renameProjectToTask.js [hunk 3]
```diff
@@ -1,24 +1,13 @@
...
-    .table('ProjectHistory')
-    .config()
-    .update({name: 'TaskHistory'})
-    .run()
-  await r
-    .table('TaskHistory')
-    .indexDrop('projectIdUpdatedAt')
-    .run()
+  await r.table('Project').config().update({name: 'Task'}).run()
+  await r.table('Task').wait()
```
Comment (mattkrick) [inline packages/server/database/migrations/20180206135107-renameProjectToTask.js:4]: moved this line up a few to give it a few more ms to kick in.
this shouldn't be necessary, but we live in a crazy world.

**Comment by mattkrick** [comment]
@Dschoordsch ready for another review

**Comment by mattkrick** [comment]
@Dschoordsch if we move `afterAll` to its own file will we always have to run tests with `--runInBand` and guarantee that `setup.ts` is run last?

**Comment by Dschoordsch** [comment]
> @Dschoordsch if we move `afterAll` to its own file will we always have to run tests with `--runInBand` and guarantee that `setup.ts` is run last?

No, it's part of the setup for each test https://github.com/ParabolInc/parabol/blob/f7d24ed7de01d02ca66936534c194c688d67056d/packages/server/jest.config.js#L22 so it will be run for each file

**Comment by mattkrick** [comment]
THAT IS AWESOME

**Pull request merged**